cumulusci/tasks/metadata/modify.py

```diff
@@ -1,52 +1,111 @@
+            "description": (
+                "A list of dictionaries containing path and xpath "
+                "keys. The path key is a file path that supports "
+                "wildcards and xpath is the xpath for the elements "
+                "to remove.  Multiple dictionaries can be passed in "
+                "the list to run multiple removal queries in the same "
+                "task.  Metadata elements in the xpath need to be prefixed "
+                "with ns:, for example: ./ns:Layout/ns:relatedLists"
+            ),
             "required": True,
         },
         "chdir": {
             "description": "Change the current directory before running the replace"
         },
     }
 
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+        self.chdir = self.options.get("chdir")
+        self.elements = self.options["elements"]
+
     def _run_task(self):
-        chdir = self.options.get("chdir")
-        if chdir:
-            self.logger.info("Changing directory to {}".format(chdir))
-        with cd(chdir):
-            for element in self.options["elements"]:
+        if self.chdir:
+            self.logger.info("Changing directory to {}".format(self.chdir))
+        with cd(self.chdir):
+            for element in self.elements:
                 self._process_element(element)
 
-    def _process_element(self, element):
+    def _process_element(self, step):
         self.logger.info(
-            "Removing elements matching {xpath} from {path}".format(**element)
+            "Removing elements matching {xpath} from {path}".format(**step)
         )
-        for f in glob.glob(element["path"]):
+        for f in glob.glob(step["path"], recursive=True):
+            self.logger.info(f"Checking {f}")
             with open(f, "rb") as fp:
                 orig = fp.read()
             root = ET.parse(f)
-            res = root.findall(
-                element["xpath"].replace(
-                    "ns:", "{http://soap.sforce.com/2006/04/metadata}"
-                )
+            res = root.xpath(
+                step["xpath"],
+                namespaces={
+                    "ns": "http://soap.sforce.com/2006/04/metadata",
+                    "re": "http://exslt.org/regular-expressions",
+                },
             )
+            self.logger.info(f"Found {len(res)} matching elements")
             for element in res:
                 element.getparent().remove(element)
-            processed = (
-                b'<?xml version="1.0" encoding="UTF-8"?>\n'
-                + ET.tostring(root, encoding="utf-8")
-                + b"\n"
-            )
+
+            processed = bytes(salesforce_encoding(root), "utf-8")
```

Comment:
Since we have python 3 now, you could just leave this as text here, and pass encoding="utf-8" when you open the file to write.

Reply:
fd28e00d92